cividiscount.php

```diff
@@ -449,15 +450,14 @@ function cividiscount_civicrm_buildAmount($pageType, &$form, &$amounts) {
       }
     }
 
-    // this seems to incorrectly set to only the last discount but it seems not to matter in the way it is used
-    if (isset($discountApplied) && $discountApplied) {
+    if (isset($discountApplied) && $discountApplied && !empty($discounts[$appliedDiscountID])) {
       if (!empty($ps['fields'])) {
         $ps['fields'] = $amounts;
         $form->setVar('_priceSet', $ps);
       }
 
       $form->set('_discountInfo', array(
-        'discount' => $discount,
```

Comment:
`$discount` is the variable which iterates in the above for loop and contains the last discount value in it which messes up the entry in the track table.